bin/oneoff/backfill_data/user_cap_status_columns

```diff
@@ -54,30 +60,21 @@ end
 batch_size = 10_000
 users.in_batches(of: batch_size) do |users_batch|
   cap_users = users_batch.
-    select(
-      :id,
-      :cap_status,
-      :cap_status_date,
-      "properties->>'$.child_account_compliance_state' AS original_cap_status",
-      "properties->>'$.child_account_compliance_state_last_updated' AS original_cap_status_date"
-    ).
     where(cap_status: nil).
     where("properties->>'$.child_account_compliance_state' IS NOT NULL").
     where("properties->>'$.child_account_compliance_state' != ''")
 
-  cap_users.find_each do |user|
-    cap_status = user.original_cap_status == 's' ? Policies::ChildAccount::ComplianceState::LOCKED_OUT : user.original_cap_status
-    cap_status_date = user.original_cap_status_date.presence && DateTime.parse(user.original_cap_status_date).utc
-
-    if options[:skip_update]
-      user.assign_attributes(cap_status: cap_status, cap_status_date: cap_status_date)
-    else
-      user.update_columns(cap_status: cap_status, cap_status_date: cap_status_date)
+  cap_users.ids.each do |user_id|
+    unless options[:skip_update]
+      User.where(id: user_id).update_all <<-SQL.squish
+        cap_status = IF(properties->>'$.child_account_compliance_state' = 's', 'l', properties->>'$.child_account_compliance_state'),
+        cap_status_date = CAST(properties->>'$.child_account_compliance_state_last_updated' AS DATETIME)
+      SQL
```

Comment:
I really don't like using `update_all` for this; is there a specific reason to do so, rather than going back to

```ruby
 User.find(user_id).update_columns(cap_status: cap_status, cap_status_date: cap_status_date)
```

Reply:
`User.where(id: user_id).update_all` achieves the same result as `User.find(user_id).update_columns` but without loading and initializing the model instance, which will lead to memory leaks due to the serialization of `properties` ( `292.28 MB  dashboard/app`)

Reply:
Ahh, gotcha. That seems reasonable; thanks for clarifying! Can you add a comment explaining that reasoning? I'd like to capture the details for anyone reading this code in the future